### PR TITLE
PP-5142 Use ErrorResponse to construct responses created by ResponseUtil

### PR DESCRIPTION
**Pull request by stephencdaly**
- Modify ResponseUtil so that most of the methods that construct an error response use the ErrorResponse model.
- Modify tests to expect a "messages" array and to assert on the "error_identifier" - which has a value of "GENERIC" for all responses currently.
- Not all responses have been converted to use ErrorResponse yet.

### CODE REVIEW

**Comment by stephencdaly** [comment]
Codacy errors are for underscores in test names, which we don't want to enforce

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java [hunk 1]
```diff
@@ -473,7 +475,8 @@ public void cannotGetCharge_WhenInvalidChargeId() {
                 .getCharge()
                 .statusCode(NOT_FOUND.getStatusCode())
                 .contentType(JSON)
-                .body(JSON_MESSAGE_KEY, is(format("Charge with id [%s] not found.", chargeId)));
+                .body(JSON_MESSAGE_KEY, contains(format("Charge with id [%s] not found.", chargeId)))
+                .body("error_identifier", Matchers.is(ErrorIdentifier.GENERIC.toString()));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: Please static import `is`

File: src/test/java/uk/gov/pay/connector/wallets/applepay/ApplePayServiceTest.java [hunk 2]
```diff
@@ -85,6 +88,7 @@ public void shouldReturnInternalServerError_ifGatewayErrors() throws IOException
 
         verify(mockedApplePayAuthoriseService).doAuthorise(externalChargeId, validData);
         assertThat(authorisationResponse.getStatus(), is(500));
-        assertThat(authorisationResponse.getEntity(), is(ImmutableMap.of("message", "oops")));
+        ErrorResponse response = (ErrorResponse)authorisationResponse.getEntity();
+        MatcherAssert.assertThat(response.getMessages(), contains("oops"));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/wallets/applepay/ApplePayServiceTest.java]: Please static import `assertThat`

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks fine — just a couple of things with imports (which can be fixed in a new commit so I don’t have to look through all 300+ lines again!)

**Comment by stephencdaly** [comment]
Thanks for the review, I've pushed a commit with those changes

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java [hunk 3]
```diff
@@ -606,7 +609,8 @@ public void shouldGetNotFoundFor_markChargeAsCaptureApproved_whenNoChargeExists(
                 .postMarkChargeAsCaptureApproved()
                 .statusCode(NOT_FOUND.getStatusCode())
                 .contentType(JSON)
-                .body(JSON_MESSAGE_KEY, is("Charge with id [i-do-not-exist] not found."));
+                .body(JSON_MESSAGE_KEY, contains("Charge with id [i-do-not-exist] not found."))
+                .body("error_identifier", Matchers.is(ErrorIdentifier.GENERIC.toString()));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: And this one

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java [hunk 4]
```diff
@@ -622,7 +626,8 @@ public void shouldGetConflictExceptionFor_markChargeAsCaptureApproved_whenNoChar
                 .postMarkChargeAsCaptureApproved()
                 .statusCode(CONFLICT.getStatusCode())
                 .contentType(JSON)
-                .body(JSON_MESSAGE_KEY, is(expectedErrorMessage));
+                .body(JSON_MESSAGE_KEY, contains(expectedErrorMessage))
+                .body("error_identifier", Matchers.is(ErrorIdentifier.GENERIC.toString()));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: And this one

File: src/test/java/uk/gov/pay/connector/wallets/googlepay/GooglePayServiceTest.java [hunk 5]
```diff
@@ -76,6 +79,7 @@ public void shouldReturnInternalServerError_ifGatewayErrors() {
 
         verify(mockedWalletAuthoriseService).doAuthorise(externalChargeId, googlePayAuthRequest);
         assertThat(authorisationResponse.getStatus(), is(500));
-        assertThat(authorisationResponse.getEntity(), is(ImmutableMap.of("message", "oops")));
+        ErrorResponse response = (ErrorResponse)authorisationResponse.getEntity();
+        MatcherAssert.assertThat(response.getMessages(), contains("oops"));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/wallets/googlepay/GooglePayServiceTest.java]: And this one

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Missed a few…

**Comment by stephencdaly** [comment]
I got all of them now

**Comment by stephencdaly** [comment]
I just force pushed to remove some unused imports

File: src/test/java/uk/gov/pay/connector/it/gatewayclient/GatewayInvalidUrlITest.java [hunk 6]
```diff
@@ -35,6 +35,6 @@ public void shouldFailCaptureWhenInvalidConnectorUrl() {
         app.getInstanceFromGuiceContainer(CardCaptureProcess.class).runCapture(1);
 
         assertLastGatewayClientLoggingEventContains("Exception for gateway url=http://gobbledygook.invalid.url");
-        Assert.assertThat(app.getDatabaseTestHelper().getChargeStatus(testCharge.getChargeId()), Matchers.is(CAPTURE_APPROVED_RETRY.getValue()));
+        Assert.assertThat(app.getDatabaseTestHelper().getChargeStatus(testCharge.getChargeId()), is(CAPTURE_APPROVED_RETRY.getValue()));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/it/gatewayclient/GatewayInvalidUrlITest.java]: Could you static import this too please (there are a few more in other files)

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Please don’t hate me

File: src/test/java/uk/gov/pay/connector/it/gatewayclient/GatewayInvalidUrlITest.java (hunk 6 above)
Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/it/gatewayclient/GatewayInvalidUrlITest.java]: (╯°□°)╯︵ ┻━┻

Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/it/gatewayclient/GatewayInvalidUrlITest.java]: sure

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**